main.go

```diff
@@ -15,25 +17,38 @@ import (
 )
 
 const (
-	// This must eventually be a default that makes sense, and overrideable by env var or flag.
-	SETTING_FILE_PATH = "./provider_settings.json"
-	RULES_FILE_PATH   = "./rule-example.yaml"
-	OUTPUT_VIOLATIONS = "./output.yaml"
+	EXIT_ON_ERROR_CODE = 3
+)
+
+var (
+	settingsFile      = flag.String("provider-settings", "provider_settings.json", "path to the provider settings")
+	rulesFile         = flag.String("rules", "rule-example.yaml", "filename or directory containing rule files")
+	outputViolations  = flag.String("output-file", "output.yaml", "place to put all the output files")
+	errorOnViolations = flag.Bool("error-on-violation", false, "exit with 3 if any violation are found will also print violations to console")
+	logLevel          = flag.Int("verbose", 9, "level for logging output")
 )
 
 func main() {
+	flag.Parse()
+
+	err := validateFlags()
+	if err != nil {
+		fmt.Printf("%v\n", err)
+		os.Exit(1)
+	}
+
 	ctx, cancelFunc := context.WithCancel(context.Background())
 
 	logrusLog := logrus.New()
 	logrusLog.SetOutput(os.Stdout)
```

Comment:
I'm curious about these command line tools that respond on stdout...should the logging be done on Stderr?

Reply:
yeah, that works. I had forgotten that the violations are written to file not stdout.